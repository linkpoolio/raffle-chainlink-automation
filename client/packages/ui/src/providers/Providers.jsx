import React from 'react'
import { Router } from 'react-router-dom'
import { Web3WalletProvider } from './'

export const Providers = ({ children, history }) => (
  <Router history={history}>
    <Web3WalletProvider>{children}</Web3WalletProvider>
  </Router>
)
