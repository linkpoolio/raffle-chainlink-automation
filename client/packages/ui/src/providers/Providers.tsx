import React from 'react'
import { Router } from 'react-router-dom'
import { createBrowserHistory } from 'history'
import {
  ConnectKitProvider,
  ConnectKitButton,
  getDefaultClient
} from 'connectkit'
import { WagmiConfig, createClient } from 'wagmi'

import { client } from '@ui/config/client'
import { Web3WalletProvider } from './'

const history = createBrowserHistory()

export const Providers = ({ children }) => (
  <Router history={history}>
    <WagmiConfig client={client}>
      <ConnectKitProvider>
        <Web3WalletProvider>{children}</Web3WalletProvider>
      </ConnectKitProvider>
    </WagmiConfig>
  </Router>
)