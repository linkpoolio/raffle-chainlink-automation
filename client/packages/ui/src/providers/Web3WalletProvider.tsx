import React from 'react'
import { ethers } from 'ethers'
import { WagmiConfig, createClient } from 'wagmi'
import { InjectedConnector } from 'wagmi/connectors/injected'
import { mainnet } from 'wagmi/chains'

const defaultChains = [mainnet]

const getProvider = () =>
  new ethers.providers.JsonRpcProvider('https://main-light.eth.linkpool.io', 1)

const connectors = () => {
  return [
    new InjectedConnector({
      chains: defaultChains,
      options: { shimDisconnect: true }
    })
  ]
}

const client = createClient({
  autoConnect: true,
  connectors,
  provider: getProvider
})

export const Web3WalletProvider = ({ children }) => (
  <WagmiConfig client={client}>{children}</WagmiConfig>
)
