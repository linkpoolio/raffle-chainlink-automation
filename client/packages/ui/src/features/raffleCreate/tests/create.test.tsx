import React from 'react'
import renderer from 'react-test-renderer'

import { Providers } from '@ui/providers'
import { RaffleCreate } from '../'

const getComponent = () => {
  const component = (
    <Providers>
      <RaffleCreate />
    </Providers>
  )

  return component
}

describe('RaffleCreate', () => {
  it('matches snapshot', () => {
    const tree = renderer.create(getComponent()).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
