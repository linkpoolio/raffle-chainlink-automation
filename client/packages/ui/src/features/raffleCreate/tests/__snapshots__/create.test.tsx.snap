// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RaffleCreate matches snapshot 1`] = `
[
  <div
    className="chakra-container css-1ew2da3"
  >
    <div
      className="css-1tr0t8b"
    >
      <h2
        className="chakra-heading css-3lum0h"
      >
        Create Raffle
      </h2>
      <p
        className="chakra-text css-19rlujp"
      >
        Create dynamic or static raffle
      </p>
    </div>
    <div
      className="css-65mrn5"
    >
      <div
        className="css-0"
      >
        <div
          className="chakra-form-control css-1kxonj9"
          role="group"
        >
          <label
            className="chakra-form__label css-g6pte"
            htmlFor="field-:r0:"
            id="field-:r0:-label"
          >
            Select Raffle Type
          </label>
          <div
            className="chakra-select__wrapper css-42b2qy"
          >
            <select
              className="chakra-select css-1trmxog"
              data-testid="select-type"
              disabled={false}
              id="field-:r0:"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              readOnly={false}
              required={false}
              value={1}
            >
              <option
                value={1}
              >
                Static
              </option>
              <option
                value={0}
              >
                Dynamic
              </option>
            </select>
            <div
              className="chakra-select__icon-wrapper css-iohxn1"
            >
              <svg
                aria-hidden={true}
                className="chakra-select__icon"
                focusable={false}
                role="presentation"
                style={
                  {
                    "color": "currentColor",
                    "height": "1em",
                    "width": "1em",
                  }
                }
                viewBox="0 0 24 24"
              >
                <path
                  d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                  fill="currentColor"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
      <div
        className="css-0"
      >
        <div
          className="chakra-form-control css-1kxonj9"
          role="group"
        >
          <label
            className="chakra-form__label css-g6pte"
            htmlFor="field-:r1:"
            id="field-:r1:-label"
          >
            Raffle Name
          </label>
          <input
            className="chakra-input css-1cim8y9"
            disabled={false}
            id="field-:r1:"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Name"
            readOnly={false}
            required={false}
            type="text"
            value=""
          />
          <div
            className="chakra-form__helper-text css-1w3rar9"
            id="field-:r1:-helptext"
          >
            Max 40 characters
          </div>
        </div>
      </div>
      <div
        className="css-0"
      >
        <div
          className="chakra-form-control css-1kxonj9"
          role="group"
        >
          <label
            className="chakra-form__label css-g6pte"
            htmlFor="field-:r2:"
            id="field-:r2:-label"
          >
            Number of winners
          </label>
          <input
            className="chakra-input css-1cim8y9"
            disabled={false}
            id="field-:r2:"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Number"
            readOnly={false}
            required={false}
            type="text"
            value={1}
          />
          <div
            className="chakra-form__helper-text css-1w3rar9"
            id="field-:r2:-helptext"
          >
            Max 120 winners
          </div>
        </div>
      </div>
      <div
        className="css-0"
      >
        <div
          className="chakra-form-control css-1kxonj9"
          role="group"
        >
          <label
            className="chakra-form__label css-g6pte"
            htmlFor="field-:r3:"
            id="field-:r3:-label"
          >
            Prize description
          </label>
          <input
            className="chakra-input css-1cim8y9"
            disabled={false}
            id="field-:r3:"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Name"
            readOnly={false}
            required={false}
            type="text"
            value=""
          />
          <div
            className="chakra-form__helper-text css-1w3rar9"
            id="field-:r3:-helptext"
          >
            Max 40 charcters
          </div>
        </div>
      </div>
      <div
        className="css-tuh9u2"
      >
        <div
          className="chakra-form-control css-1kxonj9"
          role="group"
        >
          <label
            className="chakra-form__label css-g6pte"
            htmlFor="field-:r4:"
            id="field-:r4:-label"
          >
            Participants
          </label>
          <input
            accept="text/csv, .csv, application/vnd.ms-excel"
            autoComplete="off"
            multiple={false}
            onChange={[Function]}
            onClick={[Function]}
            required={false}
            style={
              {
                "display": "none",
              }
            }
            tabIndex={-1}
            type="file"
          />
          <div
            className="css-1gng5hv"
            onBlur={[Function]}
            onClick={[Function]}
            onDragEnter={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
          >
            <p
              className="chakra-text css-0"
            >
               Drop CSV file here or click to upload
            </p>
          </div>
          <div
            className="chakra-form__helper-text css-1w3rar9"
            id="field-:r4:-helptext"
          >
             
            <button>
              Click here to download
            </button>
             CSV file example
          </div>
        </div>
      </div>
    </div>
    <div
      className="css-gmuwbf"
    >
      <button
        className="chakra-button css-4235hj"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Create
      </button>
    </div>
  </div>,
  <span />,
  <span
    hidden={true}
    id="__chakra_env"
  />,
]
`;
